frontend/drivers/platform_psp.c

```diff
@@ -351,21 +350,47 @@ static void frontend_psp_exec(const char *path, bool should_load_game)
       char *param2 = strstr(boot_params, "&param2=");
       if (param1 != NULL && param2 != NULL)
       {
-         param1 += 7;
-         memcpy(core_name, param1, param2 - param1);
-         core_name[param2-param1] = 0;
-         sprintf(argp, param2 + 8);
-         ret = sceAppMgrLoadExec(core_name, (char * const*)((const char*[]){argp, 0}), NULL);
-         RARCH_LOG("Attempt to load executable: [%d].\n", ret);
-         goto exit;
+         if (param2 > param1 && (param2 - (param1+7) < 256) && strlen(param2+8) < 512)
+         {
+            /* handle case where param2 follows param1 */
+            param1 += 7;
+            memcpy(core_name, param1, param2 - param1);
+            core_name[param2-param1] = 0;
+            sprintf(argp, param2 + 8);
+            int ret = sceAppMgrLoadExec(core_name, (char * const*)((const char*[]){argp, 0}), NULL);
+            RARCH_LOG("Attempt to load executable: [%d].\n", ret);
+            return;
+         }
+         else if (param1 > param2 && (param1 - (param2+8) < 512) && strlen(param1+7) < 256)
+         {
+            /* handle case where param1 follows param2 */
+            param2 += 8;
+            memcpy(argp, param2, param1 - param2);
+            argp[param1-param2] = 0;
+            sprintf(core_name, param1 + 7);
+            int ret = sceAppMgrLoadExec(core_name, (char * const*)((const char*[]){argp, 0}), NULL);
+            RARCH_LOG("Attempt to load executable: [%d].\n", ret);
+            return;
+         }
+         else
+         {
+            RARCH_LOG("Boot params are too long. Continue normal boot.");
+         }
       }
-      RARCH_LOG("Required boot params missing. Continue nornal boot.");
+      else
+      {
+         RARCH_LOG("Required boot params missing. Continue nornal boot.");
+      }
+      
+      int ret =  sceAppMgrLoadExec(path, args == 0 ? NULL : (char * const*)((const char*[]){argp, 0}), NULL);
```

Comment:
could you refactor code to just use this sceAppMgrLoadExec call?

I see too much duplicated code

Reply:
@frangarcj refactored to remove duplicate code